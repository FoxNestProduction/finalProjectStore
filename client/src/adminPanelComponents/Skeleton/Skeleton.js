import React from 'react';
import PropTypes from 'prop-types';
import { Box } from '@mui/material';
import {
  skeletonFormWrapper,
  skeletonImg,
  skeletonWrapper,
  skeletonInput,
  animation,
} from './styles';

const Skeleton = ({ type }) => {
  return (
    <Box sx={{
      ...skeletonWrapper,
    }}
    >
      <Box sx={{
        ...skeletonImg,
        ...animation,
      }}
      />
      <Box sx={{
        ...skeletonFormWrapper,
      }}
      >
        {type === 'dish' && (
          <>
            <Box sx={{
              ...skeletonInput,
              ...animation,
            }}
            />
            <Box sx={{
              ...skeletonInput,
              ...animation,
            }}
            />
          </>
        )}
        <Box sx={{
          ...skeletonInput,
          ...animation,
        }}
        />
        <Box sx={{
          ...skeletonInput,
          ...animation,
        }}
        />
        <Box sx={{
          ...skeletonInput,
          ...animation,
        }}
        />
        <Box sx={{
          ...skeletonInput,
          ...animation,
        }}
        />
        <Box sx={{
          ...skeletonInput,
          ...animation,
        }}
        />
      </Box>
    </Box>
  );
};

export default Skeleton;

Skeleton.propTypes = {
  type: PropTypes.string,
};

Skeleton.defaultProps = {
  type: 'partner',
};
