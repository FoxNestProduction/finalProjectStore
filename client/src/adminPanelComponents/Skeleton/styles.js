export const skeletonWrapper = {
  display: 'flex',
  flexDirection: {
    mobile: 'column',
    desktop: 'row',
  },
  gap: {
    mobile: '35px',
    desktop: '3%',
  },
};

export const skeletonImg = {
  width: '100%',
  height: '400px',
  borderRadius: '16px',
  overflow: 'hidden',
  backgroundColor: '#DADADA',
};

export const skeletonFormWrapper = {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'flex-end',
  gap: {
    mobile: '20px;',
    tablet: '30px',
    desktop: '30px',
  },
  mb: {
    mobile: '10px;',
    tablet: '9px',
    desktop: '24px',
  },
  width: '100%',
};

export const skeletonInput = {
  width: '100%',
  height: '56px',
  borderRadius: '16px',
  backgroundColor: '#DADADA',

  '&:nth-last-child(-n+3)': {
    height: '168px',
  },
};

export const animation = {
  boxShadow: '6px 71px 35px 0px rgba(229, 229, 229, 0.70)',
  background: 'linear-gradient(-75deg, transparent 50%, rgba(92, 78, 174, 1), transparent 80%) 200% 100% / 200% 100%, #EAEAEA',
  animation: 'loading 4s infinite linear',
  backgroundAttachment: 'fixed',

  '@keyframes loading': {
    'to': {
      backgroundPosition: '-200% 0',
    },
  },
};
