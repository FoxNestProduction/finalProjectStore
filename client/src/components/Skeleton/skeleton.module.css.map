{"version":3,"sources":["skeleton.module.scss","skeleton.module.css"],"names":[],"mappings":"AAAA;EACI,WAAA;EACA,aAAA;EACA,mBAAA;EACA,uBAAA;ACCJ;;ADIA;EACI,aAAA;EACA,sBAAA;EACA,eAAA;EACA,WAAA;EACA,oBAAA;EACA,sDAAA;EACA,mBAAA;EACA,yBAAA;ACDJ;;ADIA;EACI,WAAA;EACA,aAAA;EACA,mBAAA;EACA,4BAAA;ACDJ;;ADIA;EACI,gBAAA;EACA,iBAAA;EACA,UAAA;EACA,YAAA;EACA,mBAAA;ACDJ;ADGI;EAPJ;IAQQ,iBAAA;ECAN;AACF;ADCI;EAVJ;IAWQ,iBAAA;ECEN;AACF;;ADIA;EACI,aAAA;EACA,sBAAA;EACA,uBAAA;EACA,aAAA;EACA,WAAA;EAEA,oBAAA;EACA,wBAAA;EACA,sDAAA;EACA,mBAAA;EACA,yBAAA;ACFJ;ADII;EAbJ;IAcQ,gBAAA;ECDN;AACF;ADEI;EAhBJ;IAiBQ,gBAAA;ECCN;AACF;ADAI;EAnBJ;IAoBQ,gBAAA;ECGN;AACF;;ADAA;EACI,WAAA;EACA,aAAA;EACA,mBAAA;EACA,4BAAA;ACGJ;ADDI;EANJ;IAOQ,mBAAA;IACA,aAAA;ECIN;AACF;ADHI;EAVJ;IAWQ,mBAAA;IACA,aAAA;ECMN;AACF;;ADHA;EACI,iBAAA;EACA,UAAA;EACA,YAAA;EACA,mBAAA;ACMJ;ADJI;EANJ;IAOQ,iBAAA;ECON;AACF;ADNI;EATJ;IAUQ,iBAAA;ECSN;AACF;;ADNA;EACI,oBAAA;EACA,gBAAA;EACA,kBAAA;EACA,mBAAA;EACA,UAAA;EACA,gBAAA;EACA,YAAA;ACSJ;;ADJA;EACI,aAAA;EACA,sBAAA;EACA,uBAAA;EACA,aAAA;EACA,WAAA;EACA,gBAAA;EACA,oBAAA;EACA,wBAAA;EACA,sDAAA;EACA,mBAAA;EACA,yBAAA;ACOJ;ADLI;EAbJ;IAcQ,gBAAA;ECQN;AACF;ADPI;EAhBJ;IAiBQ,iBAAA;ECUN;AACF;;ADJA;EACI,aAAA;EACA,sBAAA;EACA,eAAA;EACA,WAAA;EACA,oBAAA;EACA,8CAAA;EACA,mBAAA;EACA,yBAAA;ACOJ;;ADJA;EACI,WAAA;EACA,aAAA;EACA,mBAAA;EACA,4BAAA;ACOJ;;ADJA;EACI,YAAA;EACA,gBAAA;EACA,iBAAA;EACA,cAAA;ACOJ;ADNI;EALJ;IAMQ,iBAAA;IACA,cAAA;ECSN;AACF;ADRI;EATJ;IAUQ,cAAA;ECWN;AACF;ADVI;EAZJ;IAaQ,iBAAA;IACA,cAAA;ECaN;AACF;;ADVA;EACI,UAAA;EACA,YAAA;EACA,mBAAA;EACA,gBAAA;EACA,iBAAA;ACaJ;ADZI;EANJ;IAOQ,iBAAA;ECeN;AACF;ADdI;EATJ;IAUQ,iBAAA;ECiBN;AACF;;ADdA;EACI,UAAA;EACA,YAAA;EACA,gBAAA;EACA,iBAAA;ACiBJ;ADhBI;EALJ;IAMQ,iBAAA;ECmBN;AACF;ADlBI;EARJ;IASQ,iBAAA;ECqBN;AACF;;ADfA;EACI,aAAA;EACA,sBAAA;EACA,mBAAA;EACA,uBAAA;EACA,WAAA;ACkBJ;ADjBI;EANJ;IAOQ,mBAAA;IACA,uBAAA;ECoBN;AACF;;ADjBA;EACI,WAAA;EACA,oBAAA;EACA,mBAAA;ACoBJ;ADnBI;EAJJ;IAKQ,cAAA;IACA,mBAAA;ECsBN;AACF;ADrBI;EARJ;IASQ,cAAA;IACA,mBAAA;ECwBN;AACF;;ADrBA;EACI,aAAA;EACA,sBAAA;EACA,WAAA;EACA,iBAAA;ACwBJ;;ADrBA;EACI,UAAA;EACA,YAAA;EACA,mBAAA;EACA,gBAAA;EACA,iBAAA;ACwBJ;ADvBI;EANJ;IAOQ,iBAAA;EC0BN;AACF;ADzBI;EATJ;IAUQ,iBAAA;EC4BN;AACF;;ADzBA;EACI,wBAAA;EACA,YAAA;EACA,mBAAA;EACA,iBAAA;EACA,kBAAA;EACA,gBAAA;AC4BJ;AD3BI;EAPJ;IAQQ,wBAAA;IACA,iBAAA;IACA,aAAA;EC8BN;AACF;;AD3BA;EACI,oBAAA;EACA,UAAA;EACA,gBAAA;EACA,YAAA;EACA,mBAAA;EACA,gBAAA;EACA,kBAAA;AC8BJ;AD7BI;EARJ;IASQ,kBAAA;ECgCN;AACF;AD/BI;EAXJ;IAYQ,iBAAA;IACA,gBAAA;IACA,sBAAA;ECkCN;AACF;;AD7BA;EACI,WAAA;EACA,aAAA;EACA,mBAAA;EACA,mBAAA;EACA,oDAAA;EACA,iBAAA;EACA,kBAAA;ACgCJ;;AD7BA;EACI,aAAA;EACA,mBAAA;EACA,2BAAA;ACgCJ;;AD7BA;EACI,WAAA;EACA,YAAA;EACA,kBAAA;EACA,yBAAA;ACgCJ;;AD7BA;EACI,iBAAA;EACA,UAAA;EACA,gBAAA;EACA,YAAA;ACgCJ;;AD7BA;EACI,gBAAA;EACA,YAAA;EACA,+BAAA;ACgCJ;;AD7BA;EACI,sBAAA;EACA,gBAAA;EACA,YAAA;EACA,gBAAA;ACgCJ;;AD5BA;EACI,oHAAA;EACA,qCAAA;EACA,4BAAA;AC+BJ;;AD5BA;EACI;IACI,4BAAA;EC+BN;AACF","file":"skeleton.module.css"}