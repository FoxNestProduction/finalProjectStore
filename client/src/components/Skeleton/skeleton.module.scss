.container {
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
}

// Restaurant skeleton

.restaurantCardSkeleton {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    width: 100%;
    padding-bottom: 20px;
    box-shadow: 6px 71px 35px 0px rgba(229, 229, 229, 0.70);
    border-radius: 16px;
    background-color: #DADADA;
}

.restaurantImageSkeleton {
    width: 100%;
    height: 160px;
    margin-bottom: 10px;
    border-radius: 16px 16px 0 0;
}

.restaurantChipsTitleRatingSkeleton {
    margin-top: 10px;
    margin-left: 10px;
    width: 70%;
    height: 20px;
    border-radius: 16px;

    @media (min-width: 481px) {
        margin-left: 20px;
    };
    @media (min-width: 993px) {
        margin-left: 32px;
    };
}


// Restaurants Page Skeleton

.restaurantsPageCardSkeleton {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 100%;
    max-width: 315px;
    padding-bottom: 50px;
    margin: 0 auto 50px auto;
    box-shadow: 6px 71px 35px 0px rgba(229, 229, 229, 0.70);
    border-radius: 16px;
    background-color: #DADADA;

    @media (min-width: 481px) {
        max-width: 420px;
    };
    @media (min-width: 690px) {
        max-width: 500px;
    };
    @media (min-width: 993px) {
        max-width: 800px;
    };
}

.restaurantsPageImageSkeleton {
    width: 100%;
    height: 220px;
    margin-bottom: 10px;
    border-radius: 16px 16px 0 0;

    @media (min-width: 690px) {
        margin-bottom: 20px;
        height: 268px;
    };
    @media (min-width: 993px) {
        margin-bottom: 40px;
        height: 378px;
    };
}

.restaurantsPageChipsTitleRatingSkeleton {
    margin-top: 10px;
    margin-left: 15px;
    width: 70%;
    height: 30px;
    border-radius: 16px;
    
    @media (min-width: 481px) {
        margin-left: 20px;
    };
    @media (min-width: 993px) {
        margin-left: 32px;
    };
}


// Product skeleton

.productCardSkeleton {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    width: 100%;
    padding-bottom: 20px;
    box-shadow: 0 20px 20px 0px rgba(229, 229, 229, 1);
    border-radius: 16px;
    background-color: #DADADA;
}

.productImageSkeleton {
    width: 100%;
    height: 162px;
    margin-bottom: 10px;
    border-radius: 16px 16px 0 0;
}

.productChipsTitleRatingSkeleton {
    height: 20px;
    margin-top: 10px;
    margin-left: 10px;
    max-width: 46%;
    @media (min-width: 481px) {
        margin-left: 20px;
        max-width: 44%;
    };
    @media (min-width: 690px) {
        max-width: 26%;
    };
    @media (min-width: 993px) {
        margin-left: 32px;
        max-width: 17%;
    };
}

.productRaitingSkeleton {
    width: 55%;
    height: 25px;
    margin-top: 20px;
    margin-left: 10px;
    @media (min-width: 481px) {
        margin-left: 20px;
    };
    @media (min-width: 993px) {
        margin-left: 32px;
    };
}

.productPriceSkeleton {
    width: 35%;
    height: 25px;
    margin-top: 30px;
    margin-left: 10px;
    @media (min-width: 481px) {
        margin-left: 20px;
    };
    @media (min-width: 993px) {
        margin-left: 32px;
    };
}


// Reviews skeleton

.reviewCardSkeleton {
    width: 100%;
    height: 240px;
    border-radius: 16px;
    background: #DADADA;
    box-shadow: 0px 0px 25px #0000001A;
    padding-top: 35px;
    padding-left: 35px;
}

.userContainer {
    display: flex;
    align-items: center;
    justify-content: flex-start;
}

.avatar {
    width: 40px;
    height: 40px;
    border-radius: 50%;
    background-color: #DADADA;
}

.username {
    margin-left: 20px;
    width: 60%;
    max-width: 300px;
    height: 25px;
}

.reviewText {
    margin-top: 25px;
    height: 60px;
    width: calc(100% - 20px - 33px);
}

.rating {
    width: calc(80% + 5px);
    max-width: 360px;
    height: 30px;
    margin-top: 20px;
}


.loading {
    background: linear-gradient(-75deg, transparent 50%, rgba(92, 78, 174, 1), transparent 80%) 200% 100% / 200% 100%, #EAEAEA;
    animation: loading 4s infinite linear;
    background-attachment: fixed;
}

@keyframes loading {
    to {
        background-position: -200% 0;
    }
}
