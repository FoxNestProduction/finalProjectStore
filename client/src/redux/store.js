import { configureStore, getDefaultMiddleware, combineReducers } from '@reduxjs/toolkit';
// todo: правила для redux-persist
// eslint-disable-next-line import/no-extraneous-dependencies
import storage from 'redux-persist/lib/storage';
// eslint-disable-next-line import/no-extraneous-dependencies
import { persistReducer } from 'redux-persist';
import modalSlice from './slices/modalSlice';
import productsSlice from './slices/productsSlice';
import authorizationSlice from './slices/authorizationSlice';
import userSlice from './slices/userSlice';
import partnersSlice from './slices/partnersSlice';
import searchSlice from './slices/searchSlice';
import errorSlice from './slices/errorSlice';
import reviewsSlice from './slices/reviewsSlice';
import favouriteSlice from './slices/favouriteSlice';
import cartSlice from './slices/cartSlice';
import orderSlice from './slices/orderSlice';
import filterSlice from './slices/filterSlice';
import scrollAnchorSlice from './slices/scrollAnchorSlice';
import newGoogleUserSlice from './slices/newGoogleUserSlice';

const authPersistConfig = {
  key: 'authorization',
  version: 1,
  storage,
};

const userPersistConfig = {
  key: 'user',
  version: 1,
  storage,
};

const favouritePersistConfig = {
  key: 'favorites',
  version: 1,
  storage,
};

const cartPersistConfig = {
  key: 'cart',
  version: 1,
  storage,
};

const reducer = combineReducers({
  error: errorSlice,
  modal: modalSlice,
  partners: partnersSlice,
  authorization: persistReducer(authPersistConfig, authorizationSlice),
  user: persistReducer(userPersistConfig, userSlice),
  newGoogleUser: newGoogleUserSlice,
  products: productsSlice,
  search: searchSlice,
  order: orderSlice,
  reviews: reviewsSlice,
  favourites: persistReducer(favouritePersistConfig, favouriteSlice),
  filter: filterSlice,
  cart: persistReducer(cartPersistConfig, cartSlice),
  scrollAnchor: scrollAnchorSlice,
});

const middleware = getDefaultMiddleware({
  serializableCheck: false,
});

const store = configureStore({
  reducer,
  middleware,
});

export default store;
