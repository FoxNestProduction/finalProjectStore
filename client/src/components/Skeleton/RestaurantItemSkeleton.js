import React from 'react';
import classNames from 'classnames';
import styles from './skeleton.module.scss';

const Skeleton = () => {
  return (
    // <Box sx={flexCenter}>
    //   <Card sx={card}>
    //     <Box sx={{ ...imageSkeleton, ...loading }} />
    //     <Box sx={{ ...chipsTitleRatingSkeleton, ...loading }} />
    //     <Box sx={{ ...chipsTitleRatingSkeleton, ...loading }} />
    //     <Box sx={{ ...chipsTitleRatingSkeleton, ...loading }} />
    //   </Card>
    // </Box>
    <div className={styles.container}>
      <div className={styles.card}>
        <div className={classNames(styles.imageSkeleton, styles.loading)} />
        <div className={classNames(styles.chipsTitleRatingSkeleton, styles.loading)} />
        <div className={classNames(styles.chipsTitleRatingSkeleton, styles.loading)} />
        <div className={classNames(styles.chipsTitleRatingSkeleton, styles.loading)} />
      </div>
    </div>
  );
};

export default Skeleton;
