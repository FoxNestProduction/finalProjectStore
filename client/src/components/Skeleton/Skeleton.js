import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import styles from './skeleton.module.scss';

const Skeleton = ({ skeletonType }) => {
  return (
    <div className={styles.container}>
      { skeletonType === 'restaurant' ? (
        <div className={styles.restaurantCardSkeleton}>
          <div className={classNames(styles.restaurantImageSkeleton, styles.loading)} />
          <div
            className={classNames(styles.restaurantChipsTitleRatingSkeleton, styles.loading)}
          />
          <div
            className={classNames(styles.restaurantChipsTitleRatingSkeleton, styles.loading)}
          />
          <div
            className={classNames(styles.restaurantChipsTitleRatingSkeleton, styles.loading)}
          />
        </div>
      ) : skeletonType === 'restaurantsPage' || skeletonType === 'oneRestaurantPage' ? (
        <div className={skeletonType !== 'oneRestaurantPage' ? styles.restaurantsPageCardSkeleton : styles.oneRestaurantPageCardSkeleton}>
          <div className={classNames(styles.restaurantsPageImageSkeleton, styles.loading)} />
          <div
            className={classNames(styles.restaurantsPageChipsTitleRatingSkeleton, styles.loading)}
          />
          <div
            className={classNames(styles.restaurantsPageChipsTitleRatingSkeleton, styles.loading)}
          />
          <div
            className={classNames(styles.restaurantsPageShowMoreSkeleton, styles.loading)}
          />
        </div>
      ) : skeletonType === 'product' ? (
        <div className={styles.productCardSkeleton}>
          <div className={classNames(styles.productImageSkeleton, styles.loading)} />
          <div className={classNames(styles.productChipsTitleRatingSkeleton, styles.loading)} />
          <div className={classNames(styles.productChipsTitleRatingSkeleton, styles.loading)} />
          <div className={classNames(styles.productRaitingSkeleton, styles.loading)} />
          <div className={classNames(styles.productPriceSkeleton, styles.loading)} />
        </div>
      ) : skeletonType === 'oneProductPage' ? (
        <div className={styles.oneProductPageCardSkeleton}>
          <div className={classNames(styles.oneProductPageImageSkeleton, styles.loading)} />
          <div className={styles.oneProductPageDescriptionWrapper}>
            <div className={classNames(
              styles.oneProductPageRaitingSkeleton,
              styles.loading,
            )}
            />
            <div className={classNames(
              styles.oneProductPageRaitingSkeleton,
              styles.loading,
            )}
            />
            <div className={classNames(styles.oneProductPageDescription, styles.loading)} />
            <div className={classNames(styles.oneProductPagePriceSkeleton, styles.loading)} />
          </div>
        </div>
      ) : (
        <div className={styles.reviewCardSkeleton}>
          <div className={styles.userContainer}>
            <div className={classNames(styles.avatar, styles.loading)} />
            <div className={classNames(styles.username, styles.loading)} />
          </div>
          <div className={classNames(styles.reviewText, styles.loading)} />
          <div className={classNames(styles.rating, styles.loading)} />
        </div>
      ) }
    </div>
  );
};

export default Skeleton;

Skeleton.propTypes = {
  skeletonType: PropTypes.string,
};

Skeleton.defaultProps = {
  skeletonType: '',
};
