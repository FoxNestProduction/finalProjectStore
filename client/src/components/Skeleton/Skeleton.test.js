import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Skeleton from './Skeleton';

describe('Skeleton component', () => {
  test('Restaurant skeleton snapshot', () => {
    const { asFragment } = render(
      <Skeleton skeletonType="restaurant" />,
    );

    const card = screen.getByTestId('restaurant');
    expect(asFragment()).toMatchSnapshot();
    expect(card).toBeInTheDocument();
  });

  test('OneProductPage skeleton snapshot', () => {
    render(
      <Skeleton skeletonType="oneProductPage" />,
    );

    const card = screen.getByTestId('oneProductPage');
    expect(card).toBeInTheDocument();
  });
});
