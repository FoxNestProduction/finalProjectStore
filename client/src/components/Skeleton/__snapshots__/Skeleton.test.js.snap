// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton component Restaurant skeleton snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="restaurantCardSkeleton"
      data-testid="restaurant"
    >
      <div
        class="restaurantImageSkeleton loading"
      />
      <div
        class="restaurantChipsTitleRatingSkeleton loading"
      />
      <div
        class="restaurantChipsTitleRatingSkeleton loading"
      />
      <div
        class="restaurantChipsTitleRatingSkeleton loading"
      />
    </div>
  </div>
</DocumentFragment>
`;
